proto/rpc/v1/amount.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package rpc.v1;
+
+message CurrencyAmount {
```

Comment:
In the current API, when the amount is native (XRP), the amount is specified with just a string (which represents the number of drops, and is really a 64 bit unsigned integer), as opposed to some sort of object. Non-native currency amounts are specified with an object that has fields for the currency code, value (could be in scientific notation, not necessarily an integer), and issuer. @keefertaylor are you proposing to deviate from the current API, and to specify all currency amounts, including XRP, as an IssuedCurrencyAmount object?

Reply:
I think we should consider it. The current API surface is just shorthand for known values (if the issuer is omitted, then the issuer is address 0).

Reply:
Well, if we get rid of XRPAmount, and just use (what will be called) IssuedCurrencyAmount, the value field is a string, because issued currencies can be use scientific notation, or floating point, whereas XRPAmount is always uint64. I like the idea of getting rid of XRPAmount, for consistency, but I don't like the idea of losing the type information.

Reply:
@keefertaylor what do you think we should do about the type of 'value' being a string? XRPAmount specifies drops as uint64, and I'm hesitant to lose that type information, if we merge XRPAmount into IssuedCurrencyAmount.

Reply:
I think an unfortunate reality is that if we can't bound integers to 64 bits (AFAIK we cannot) then we have to use a string. This seems to be how the JSON API worked too. 

It just happens to be a known property of XRP that drops can fit into a unsigned 64 bit integer.

Reply:
@keefertaylor @MarkusTeufelberger I decided to keep XRP amounts as their own type (though I renamed it to `XRPDropsAmount`). I prefer using uint64 for drops as opposed to just a string. Types are good. lmk what you think